### PR TITLE
http basic auth built in

### PR DESCRIPTION
**Pull request by adrianmroz-allegro**
- cluster auth settings in cluster configuration
- compose request decorator with cluster auth headers
- update docs

### CODE REVIEW

File: docs/configuration-cluster.md [hunk 1]
```diff
@@ -169,6 +169,25 @@ This will put additional load on the data store but will ensure that dimension a
 
 How often should source schema be reloaded in ms. Default value of 0 disables periodical source refresh.
 
+**auth**
```
Comment (mkuthan) [inline docs/configuration-cluster.md:172]: I would put this section just after "url" or "host" section. Now the authentication section is together with sourceReintrospectInterval - strange.

**Review by mkuthan — APPROVED** [review]
Looks good, a few minor comments.

File: docs/example/request-decoration/druid-request-decorator.js [hunk 2]
```diff
@@ -27,8 +20,7 @@ exports.druidRequestDecoratorFactory = function (logger, params) {
   return function (decoratorRequest) {
     const decoration = {
       headers: {
-        "Authorization": auth,
-        "X-I-Like": "Koalas",
+        "X-I-Like": like,
```
[the remark this answers is not in the document]
Comment (adrianmroz-allegro) [reply docs/example/request-decoration/druid-request-decorator.js:23]: I don't understand. What you propose to show in example if not adding header? Request decorator is about changing requests, adding header is the simplest example.

[the remark this answers is not in the document]
Comment (mkuthan) [reply docs/example/request-decoration/druid-request-decorator.js:23]: You are totally right

File: src/common/models/cluster/cluster.mocha.ts [hunk 3]
```diff
@@ -146,7 +184,8 @@ describe("Cluster", () => {
         version: "new-version",
         type: "druid",
         requestDecorator: null,
-        retry: new RetryOptions()
+        retry: new RetryOptions(),
+        auth: undefined
```
[the remark this answers is not in the document]
Comment (mkuthan) [reply src/common/models/cluster/cluster.mocha.ts:188]: But test for retrying could check only relevant part of the output and ignore everything else (like auth field).

**Pull request merged**